lsp-mode.el

```diff
@@ -1229,7 +1229,7 @@ Symlinks are not followed."
   "Return t if PATH-A is ancestor of PATH-B.
 Symlinks are not followed."
   (unless (lsp-f-same? path-a path-b)
-    (s-prefix? (lsp-f-canonical path-a)
+    (s-prefix? (concat (lsp-f-canonical path-a) "/")
```

Comment:
"/" is os specific.

Reply:
Is it? The `lsp-f-canonical` is already normalizing path to use `/` already.
I've tested on Windows and Ubuntu, it seems working fine. Of course I can try to use `f--path-separator` too, but it seems to be over kill?

Reply:
can you then add tests illustrating how it works with windows paths?

Reply:
I've added tests and also change to use `f--path-separator` too

Reply:
Revert the new test cases, it will not pass since the CI machine is in Ubuntu. Letting CI running on Windows is a pain so ...

Reply:
> Revert the new test cases, it will not pass since the CI machine is in Ubuntu. Letting CI running on Windows is a pain so ...

It is still fine to have them wrapped conditionally to run only on windows. It will be good for documentation purpose.